src/app/common/user.service.spec.js

```diff
@@ -0,0 +1,159 @@
+      });
+
+      it('most init values are falsy', function() {
+        expect(userService.user.BMI).toBeNull();
+        expect(userService.user.BMR).toBeNull();
+        expect(userService.user.dailyCalories).toBeNull();
+        expect(userService.user.sleepDuration).toBeNull();
+
+        _.values(userService.user.sleep).forEach(function (value) {
+          expect(value).toBeFalsy();
+        });
+        _.values(userService.user.activity).forEach(function (value) {
+          expect(value).toBeFalsy();
+        });
+      });
+      it('metrics init values are true', function() {
+        expect(userService.user.nutrition.isWeightKg).toBeTruthy();
+        expect(userService.user.nutrition.isHeightCm).toBeTruthy();
+      });
+    });
+
+    describe('calculate BMI', function() {
+      it('is function', function() {
+        expect(userService.calculateBMI).toEqual(jasmine.any(Function));
+      });
+      it('check formula', function() {
+        userService.user.nutrition.weight = 40;
+        userService.user.nutrition.height = 150;
+        expect(userService.calculateBMI()).toEqual(17.8);
+      });
+      it('Undefined without any input', function() {
+        expect(userService.calculateBMI()).toBeUndefined();
+        userService.user.nutrition.height = 150;
+        expect(userService.calculateBMI()).toBeUndefined();
+        userService.user.nutrition.weight = 40;
+        expect(userService.calculateBMI()).toBeDefined();
+      });
+    });
+
+    describe('calculate BMR', function() {
+      it('is function', function() {
+        expect(userService.calculateBMR).toEqual(jasmine.any(Function));
+      });
+      it('undefined without inputs', function() {
+        expect(userService.calculateBMR()).toBeUndefined();
+      });
+      it('check formula for man', function() {
+        userService.user.nutrition.weight = 40;
+        expect(userService.calculateBMR()).toBeUndefined();
+        userService.user.nutrition.height = 150;
+        expect(userService.calculateBMR()).toBeUndefined();
+        userService.user.age = 25;
+        expect(userService.calculateBMR()).toBeUndefined();
+
+        userService.user.isMale = true;
+        expect(userService.calculateBMR()).toEqual(1198);
+      });
+      it('check formula for woman', function() {
+        userService.user.nutrition.weight = 40;
+        userService.user.nutrition.height = 150;
+        userService.user.age = 25;
+        userService.user.isMale = false;
+
+        expect(userService.calculateBMR()).toEqual(1208);
+      });
+    });
+    describe('sleepDuration', function() {
+      it('is a function', function() {
+        expect(userService.getUser).toEqual(jasmine.any(Function));
+      });
+      it('returns null without properties defined', function() {
+        expect(userService.sleepDuration()).toBeNull();
+        userService.sleepDuration();
+        expect(userService.sleepDuration()).toBeNull();
```

Comment:
- [ ] rewrite

I would appreciate you leaving the action that you expect from me regarding comment.
For example: 
- Explain 
- Rewrite
- Refactor
- Read this and rewrite